scalafix/rules/src/main/scala/rsc/rules/semantics/Scopes.scala

```diff
@@ -8,24 +8,35 @@ import scala.meta.internal.{semanticdb => s}
 import scala.meta.internal.semanticdb.Scala._
 import scala.meta.internal.semanticdb.Scala.{Descriptor => d}
 import scala.meta.internal.semanticdb.Scala.{Names => n}
+import scala.meta.internal.semanticdb.SymbolInformation
 
 sealed trait Scope {
-  def lookup(name: n.Name): String
 
-  protected def member(symtab: Symtab, sym: String, name: n.Name): String = {
-    val info = name match {
+  val symtab: Symtab
+
+  val sym: String
+
+  def lookupInfo(name: n.Name): Option[SymbolInformation]
+
+  def lookup(name: n.Name): String =
+    lookupInfo(name).map(_.symbol).getOrElse(Symbols.None)
+
+  protected def memberInfo(name: n.Name): Option[SymbolInformation] = {
+    name match {
       case n.TermName(value) =>
         val packageMember = symtab.info(Symbols.Global(sym, d.Package(value)))
         val termMember = symtab.info(Symbols.Global(sym, d.Term(value)))
         packageMember.orElse(termMember)
       case n.TypeName(value) =>
         symtab.info(Symbols.Global(sym, d.Type(value)))
     }
-    info.map(_.symbol).getOrElse(Symbols.None)
   }
+
+  protected def member(name: n.Name): String =
+    memberInfo(name).map(_.symbol).getOrElse(Symbols.None)
 }
 
-case class ImporterScope(symtab: Symtab, sym: String, importees: List[Importee]) extends Scope {
+final case class ImporterScope(symtab: Symtab, sym: String, importees: List[Importee]) extends Scope {
```

Comment:
I have mixed feelings about `final`. On the one hand, it's a best practice. On the other hand, when you seriously buy into it, you kinda have to write it everywhere, which introduces noticeable boilerplate. Let's discuss universal adoption of `final` a little bit later when I have more time (I made a note to follow up on that), and in the meanwhile feel free to sprinkle it around.